### ISSUE TITLE
[Android] Touch is 1-2 frames late

### ISSUE THREAD

**Issue by Rene-Damm** [Unity-Technologies/InputSystem#220]
Looks like touch input on `Touchscreen` is 1-2 frames late compared to `UnityEngine.Input`. Might be a threading issue due to the UI/app thread split on Android in combination with the background input queue.

Forum thread: https://forum.unity.com/threads/is-the-touch-screen-working-yet.544878/

**Comment by renaldasrep** [comment]
Trying to reproduce this

**Comment by renaldasrep** [comment]
Made a repro project using package 07908c7
and Unity 2018.2.5f1 (it also reproes on 2018.3.0a10)

Reproduced on:
Samsung Galaxy S8  OS:8.0.0
iPhone 8+ OS:11.4.1 (reproes but not as consistently as on android)

Old Input registers (using EventTrigger component) as expected

New Input LowLevel event registers as expected
New Input activeTouches.Count > 0 on Update registers as expected
**New Input Touchscreen InputAction.performed callback has 1-2 frame delay**

![player-unityplayeractivity-08232018124900](https://user-images.githubusercontent.com/39190411/44518526-242c6680-a6d3-11e8-8ed1-644e8b4e854d.png)

[TouchDelayRepro.zip](https://github.com/Unity-Technologies/InputSystem/files/2313891/TouchDelayRepro.zip)

**Comment by todi1856** [comment]
I am actually getting the opposite, old input is by frame latter than new one. Will continue to investigate, but this maybe some threading issue.

**Comment by renaldasrep** [comment]
What I'm currently getting is:

Old input is late on Android
New input is late on iOS

**Comment by douglas-piconi** [comment]
I wrote a quick stress test and interesting result.  Turns out you don't have to stress this at all to repro the problem.   I wrote a test to run over about 500 frames and send one piece of data to a device and read it at the end of each frame.   With 1 device.  works great over 500 frames.   I was originally going to stress a few hundred devices,  however,  it turns you you only have to a a second device.  The minute you add a second device, it fails on the first frame.  There is a 2 frame delay for the data on both devices.
